mdata/notifier.go

```diff
@@ -83,7 +84,10 @@ func (cl Notifier) Handle(data []byte) {
 		}
 
 		// get metric
-		if agg, ok := cl.Metrics.Get(ms.Key); ok {
+		if cl.CreateMissingMetrics {
+			agg := cl.Metrics.GetOrCreate(ms.Key)
```

Comment:
If `GetOrCreate` wouldn't directly acquire a write lock instead of first checking whether the metric needs to be created while only holding a read lock then in situations like this one we could just call `cl.Metrics.GetOrCreate()` without this whole "if the metric exists do this or otherwise that"